spec/connectors/crawler/crawler_scheduler_spec.rb

```diff
@@ -156,6 +162,70 @@
         # it will return the base scheduling
         it_behaves_like 'triggers', nil
       end
+
+      context 'when base and custom scheduling are enabled and are scheduled after the poll interval' do
+        let(:sync_enabled) { true }
+        let(:weekly_enabled) { true }
+        let(:monthly_enabled) { true }
+
+        let(:next_trigger_time) { time_at_poll_start + poll_interval + 10 }
+        let(:weekly_next_trigger_time) { time_at_poll_start + poll_interval + 10 }
+        let(:monthly_next_trigger_time) { time_at_poll_start + poll_interval + 10 }
+
+        before(:each) do
+          allow(cron_parser).to receive(:next_time).with(time_at_poll_start).and_return(next_trigger_time, weekly_next_trigger_time, monthly_next_trigger_time)
+          expect(Utility::Logger).to receive(:debug).exactly(3).times.with(instance_of(String)) # expect three debug messages because three schedules are not being triggered
```

Comment:
I think it's more important to verify that the debug log we receive is the expected one. Because there's three we can use match and a simple regex, e.g. `match(/some_pattern/)`. I don't think we need the full string matched, just the beginning or end so we're sure it's the expected log.

Also (total nit) if the comment goes above the line it's much shorter and will be kind to IDEs.

```suggestion
          # expect three debug messages because three schedules are not being triggered
          expect(Utility::Logger).to receive(:debug).exactly(3).times.with(match(/^Sync for \w+ not triggered as .*/))
```

Reply:
Good callout! I had to add a tiny bit more to the regex (in the form of an OR) to get it to match on slight variation in the `identifier` name the test kicks out, but otherwise 100% agree here.

`^Sync for (\w+.*)|( - \w+) not triggered as .*`

```
Sync for Formatted not triggered as 2025-01-10 09:39:44....etc
Sync for Formatted - weekly not triggered as 2025-01-10 09:39:44...etc
```